### DISCUSSION TITLE
How to configure Copilot to automatically enable when nvim starts?

### DISCUSSION THREAD

**Discussion by joehdodd** [zbirenbaum/copilot.lua#261] [Q&A] [answered]
Sorry if I've missed this in the documentation, but I was wondering how I can configure this plugin to automatically enable Copilot when I run `nvim`. Currently, every time I run `nvim` for the first time I then have to run the `Copilot auth` command manually in order to enable Copilot and get suggestions. If I don't manually run `Copilot auth`, running `Copilot` tells me that it isn't enabled and I can see that the plugin isn't loaded if I check my plugin status in the Lazy window. 

Here's my current configuration...

```lua
{
    "zbirenbaum/copilot.lua",
    cmd = "Copilot",
    build = ':Copilot auth',
    event = "InsertEnter",
    config = function()
        require("copilot").setup({
          suggestion = {
            auto_trigger = true,
            debounce = 100,
            keymap = {
              accept = "<C-l>",
            },
          }
        })
    end
}
```

**Comment by MunifTanjim** [comment, marked as answer]
You've set `cmd = "Copilot",` and `event = "InsertEnter"`.

So copilot will either load when:
- You run a `:Copilot` command
- Or you've opened a file and entered Insert mode

If you don't wanna lazy load copilot, remove those 2 options, then copilot will start as soon as neovim is open.

Also copilot server will only run when you've actually opened a file. Otherwise, it'll just wait for you to open a file.

**Reply by joehdodd** [reply]
Ah, I figured it was just something I glossed over in the docs. Thank you!
